mlf_core/create/templates/mlflow/mlflow_pytorch/{{ cookiecutter.project_slug }}/{{ cookiecutter.project_slug_no_hyphen }}/{{ cookiecutter.project_slug_no_hyphen }}.py

```diff
@@ -1,95 +1,86 @@
-import os
-import mlflow
-import mlflow.pytorch
-import click
+#
+# Trains an MNIST digit recognizer using PyTorch Lightning,
+# and uses Mlflow to log metrics, params and artifacts
+# NOTE: This example requires you to first install
+# pytorch-lightning (using pip install pytorch-lightning)
+#       and mlflow (using pip install mlflow).
+#
+# pylint: disable=arguments-differ
+# pylint: disable=unused-argument
+# pylint: disable=abstract-method
+import pytorch_lightning as pl
 import torch
-import torch.optim as optim
-import time
-import tempfile
-
-from tensorboardX import SummaryWriter
-from rich import traceback
-
-from mlf_core.mlf_core import log_sys_intel_conda_env, set_general_random_seeds
-from model.model import create_model, create_parallel_model
-from training.train import train, test
-from data_loading.data_loader import load_train_test_data
-
+from argparse import ArgumentParser
+from pytorch_lightning.callbacks import ModelCheckpoint
+import mlflow
+from data_loading.data_loader import MNISTDataModule
+from model.model import LightningMNISTClassifier
+from mlf_core.mlf_core import log_sys_intel_conda_env, set_pytorch_random_seeds, set_general_random_seeds
+import os
+from rich import print
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description="PyTorch Autolog Mnist Example")
+    parser.add_argument(
+        "--general-seed",
+        type=int,
+        default=0,
+        help="number of workers (default: 3)",
+    )
+    parser.add_argument(
+        "--pytorch-seed",
+        type=int,
+        default=0,
+        help="number of workers (default: 3)",
+    )
+    parser.add_argument(
+        "--log-interval",
+        type=int,
+        default=100,
+        help="log interval (default: 100)",
+    )
 
-@click.command()
-@click.option('--cuda', type=click.Choice(['True', 'False']), default='True', help='Enable or disable CUDA support.')
-@click.option('--epochs', type=int, default=5, help='Number of epochs to train')
-@click.option('--general-seed', type=int, default=0, help='General Python, Python random and Numpy seed.')
-@click.option('--pytorch-seed', type=int, default=0, help='Pytorch specific random seed.')
-@click.option('--log-interval', type=int, default=100, help='Number of batches before logging training status')
-@click.option('--training-batch-size', type=int, default=64, help='Input batch size for training')
-@click.option('--test-batch-size', type=int, default=1000, help='Input batch size for testing')
-@click.option('--learning-rate', type=float, default=0.01, help='Learning rate')
-def start_training(cuda, epochs, general_seed, pytorch_seed, log_interval,
-                   training_batch_size, test_batch_size, learning_rate):
     # Set GPU settings
-    use_cuda = (True if cuda == 'True' else False) and torch.cuda.is_available()
+    # click option cuda
```

Comment:
Don't think that we'll need this at all?

Reply:
Well it nicely prints out how many GPUs are used and is consistent with the other templates, but pytorch lightning provides this info anyways, right?

Reply:
It should, but I did not look it up

Reply:
Then let's keep it and I will remove it if it gets redundant.